src/main/java/seedu/address/model/AddressBook.java

```diff
@@ -183,6 +184,27 @@ public boolean hasTutorial(Tutorial tutorial) {
         return tutorials.contains(tutorial);
     }
 
+    /**
+     * Returns true if a tutorial with the name {@code tutorialName} exists.
+     */
+    public boolean hasTutorialWithName(TutorialName tutorialName) {
+        requireNonNull(tutorialName);
+        try {
+            tutorials.getTutorialMatch(tutorialName);
+            return true;
+        } catch (InvalidTutorialException e) {
+            return false;
+        }
+    }
+
+    /**
+     * Returns a {@code Tutorial} with the name {@code tutorialName}. Used together with
+     * {@see hasTutorialWithName} to check if the tutorial exists.
+     */
+    public Tutorial getTutorialMatch(TutorialName tutorialName) {
```

Comment:
Would it be clearer to rename this as getTutorialWithName?

Reply:
I think i will leave it as it is for now, will update if it gets confusing later on